### ISSUE TITLE
Too Many Requests

### ISSUE THREAD

**Issue by dlindegaard** [CalinaCristian/source-from-zeit-deployment#4]
Hi.

Thank you for the tool.

I am experiencing the following error, which occurs during the download process. Some of the files are downloaded.

statusCode: 429,
statusMessage: 'Too Many Requests'
Error while downloading files. Exiting...

I believe it can be fixed by having a duration of time between each request or by having only a limited number of request going at a time.

Would you be able to make this adjustment?

**Comment by CalinaCristian** [comment]
Hi @dlindegaard ,

I'll look into it, it might be related to them moving to Vercel and [limiting rates](https://vercel.com/docs/api#api-basics/rate-limits) for [non paying users](https://vercel.com/docs/v2/platform/limits/#general-limits). I'll update the api calls anyway when I get the time.

Thank you.

**Comment by dlindegaard** [comment]
Thank you @CalinaCristian . I am a paying user if Vercel (Pro), so that is not the case. I did do a small edit of your codebase in which I made a 200ms delay between each file download request, which enabled me to complete the download of the entire project. It was, however, just a quick fix. My suggestion would be to keep track of the number of ongoing requests and ensure that it keeps below a max. I am not exactly certain what the max value actually is though.
